boards/nxp/mimxrt1180_evk/mimxrt1180_evk.dtsi

```diff
@@ -95,7 +95,7 @@
 };
 
 &switch_port0 {
-	local-mac-address = [00 00 00 01 04 00];
+	zephyr,random-mac-address;
```

Comment:
This board has Ethernet sockets that have factory MAC address marked to them. Why can't we use the factory set MAC addresses here?

Reply:
I may need to have some investigation on how to do that in both zephyr and platform.
Not sure if any fuse related driver needed?
Could you share any factory set MAC address case support in zephyr?
I'd like to support that seperately for NETC and RT1180 platform in another PR if it's okay.
Thank you very much.

Reply:
For example NXP frdm_rw612 board has both Wi-Fi and Ethernet. The Wi-Fi is properly using factory set MAC address, but the Ethernet is not. I created an issue for it #88336

I have STM nucleo_f767zi board which uses device id by default a mac address. https://github.com/zephyrproject-rtos/zephyr/blob/1e1f94405f138e5e0164dea0bcac400517ab88d5/drivers/ethernet/eth_stm32_hal.c#L846

Reply:
Having a separate PR for setting MAC address to fixed value is ok. I was just wondering why NXP is using random MAC address for several boards I have been using.

Reply:
I have had some investigation but I have no idea on the reason:(
It looks like the way getting the factory mac address varied from soc/board.
I think for i.MXRT it's from fuse which may read via `hal/nxp/mcux/mcux-sdk-ng/drivers/ocotp` driver.
For some i.MX8 and i.MX9, it's from fuse too, but it's more complicated to read. As the reading must be via talking with EdgeLock Enclave. I can see below drivers in u-boot for that.
```
drivers/misc/imx_ele/fuse.c
drivers/misc/imx_ele/ele_api.c
```
Not sure about other series SoC.
Anyway, I will take some time to make this clear, and try to support in zephyr.
Thanks.